cas-server-core-tickets/src/main/java/org/jasig/cas/ticket/AbstractTicket.java

```diff
@@ -151,12 +153,29 @@ protected boolean isExpiredInternal() {
     }
 
     @Override
-    public final int hashCode() {
-        return new HashCodeBuilder(13, 133).append(this.getId()).toHashCode();
+    public final String toString() {
+        return this.getId();
     }
 
     @Override
-    public final String toString() {
-        return this.getId();
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof Ticket)) {
+            return false;
+        }
+
+        final Ticket that = (Ticket) o;
+
+        return new EqualsBuilder()
+                .append(this.getId(), that.getId())
+                .append(this.getCreationTime(), that.getCreationTime())
+                .append(this.getGrantingTicket(), that.getGrantingTicket()).isEquals();
+    }
+
+    @Override
+    public int hashCode() {
+        return new HashCodeBuilder(17, 37).append(getId()).append(getCreationTime()).append(getGrantingTicket()).toHashCode();
```

Comment:
Do not change the sematics of the hashcode builder. This does not belong in this PR.